packages/framework/src/Model/Order/Watcher/TransportAndPaymentWatcher.php

```diff
@@ -283,4 +289,21 @@ protected function getRememberedPaymentPrices(): array
     {
         return $this->getRememberedTransportAndPayment()[self::SESSION_PAYMENT_PRICES];
     }
+
+    /**
+     * @param mixed $value
+     * @param string $methodName
+     * @return string
+     */
+    protected function getUnexpectedTypeOfPriceInSessionDeprecationMessage($value, string $methodName): string
+    {
+        $message = sprintf(
+            'Method "%s" expected the price to be saved as "%s" in session, got "%s" instead.',
+            $methodName,
+            Money::class,
+            \is_object($value) ? \get_class($value) : \gettype($value)
+        );
+
+        return $message . ' The method warned about the possibility of changed price, but it would fail in v8.0.0.';
```

Comment:
I'm not sure about mentioning concrete version... If this sentence isn't here, it would be still valuable and we'll decide later when exactly we want to remove the deprecation

Reply:
I think we'd want to remove this as soon as possible, which is `v8.0.0`. Do you think we could decide otherwise later?